### PR TITLE
chore(feature-flag): only return truthy values for onFeatureFlag

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

- v3 decide endpoint returns _all_ feature flags. Make sure flags that are returned on the callback function are only truthy

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers

### CODE REVIEW

File: src/posthog-featureflags.ts [hunk 1]
```diff
@@ -279,7 +279,16 @@ export class PostHogFeatureFlags {
         if (this.instance.decideEndpointWasHit) {
             const flags = this.getFlags()
             const flagVariants = this.getFlagVariants()
-            callback(flags, flagVariants)
+
```
Comment (neilkakkar) [inline src/posthog-featureflags.ts:282]: the `featureFlagsHandlers` expect a similar schema, so you'll need to update `receivedFeatureFlags` as well. Worth creating this into a function like `prepareFeatureFlagsForCallbacks`

Comment (neilkakkar) [reply src/posthog-featureflags.ts:282]: and corresponding test too 😅

**Review by neilkakkar — APPROVED** [review]
Missing in one place, otherwise LGTM!

File: src/posthog-core.ts [hunk 2]
```diff
@@ -997,7 +997,7 @@ export class PostHog {
      * @param {Function} [callback] The callback function will be called once the feature flags are ready or when they are updated.
      *                              It'll return a list of feature flags enabled for the user.
      */
-    onFeatureFlags(callback: (flags: string[], variants: Record<string, boolean | string>) => void): void {
+    onFeatureFlags(callback: (flags: string[], variants: Record<string, JsonType>) => void): void {
```
Comment (neilkakkar) [inline src/posthog-core.ts:1000]: variants can't be jsontype, can they?

Comment (EDsCODE) [reply src/posthog-core.ts:1000]: good catch this was bad typing while thinking i should pass through payloads

**Pull request merged**